### PR TITLE
Bump kubevirt to v0.34.2

### PR DESCRIPTION
**Pull request by RamLavi**
This PR bumps kubevirt to v0.34.2.

**What this PR does / why we need it**:

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/pool-manager/pool_test.go [hunk 1]
```diff
@@ -289,6 +289,27 @@ var _ = Describe("Pool", func() {
 			Expect(exist).To(BeFalse())
 		})
 		Describe("Update vm object", func() {
+			It("should preserve disk.io configuration on update", func() {
```
Comment (qinqon) [inline pkg/pool-manager/pool_test.go:292]: Is this related to the bumping ?

**Review by qinqon — COMMENTED** [review]
Looks good, just a question

File: pkg/pool-manager/pool_test.go (hunk 1 above)
Comment (RamLavi) [reply pkg/pool-manager/pool_test.go:292]: yep.
when the namespace is opted-in, the webhook [gets](https://github.com/k8snetworkplumbingwg/kubemacpool/blob/master/pkg/webhook/virtualmachine/virtualmachine.go#L132) the vm instance, but using a old version of kubvirt (0.29) that doesn't hold disk.io configuration.
so the disk.io conf is implicitly set to nil, and since it is set to `omitempty` then it disappears.
bumping the kubevirt to a version that has this conf fixes this, and this test makes sure it is indeed configured.

**Comment by RamLavi** [comment]
/hold

need to understand if this PR is needed.

File: pkg/pool-manager/pool_test.go (hunk 1 above)
Comment (RamLavi) [reply pkg/pool-manager/pool_test.go:292]: not really. will issue in separate PR
